compiler-core/src/type_/tests.rs

```diff
@@ -1331,12 +1331,15 @@ fn unknown_label() {
     );
 }
 
+// https://github.com/gleam-lang/gleam/issues/1669
 #[test]
 fn todo_warning_test() {
     assert_warning!(
-        "fn main() { 1 == todo }",
+        "fn main() {
+          1 == todo
+        }",
         Warning::Todo {
-            location: SrcSpan { start: 17, end: 21 },
+            location: SrcSpan { start: 27, end: 31 },
```

Comment:
I think this file has the typechecker tests, where the location was still correct. It was being lost sometime later, I don't think this test is calling `to_diagnostic` anywhere? Maybe there are some other tests for the LSP

Reply:
I have fixed the location with the change here I think https://github.com/gleam-lang/gleam/pull/1670/files#diff-69d4f3e128a4feb5c69856b9c1754c7d6cc0a2be8a9fcaa7e7b5e682cb965a84L33

But the tests do not test for the warning string to reflect the location.

before:

<img width="1403" alt="Screen Shot 2022-06-25 at 19 08 06" src="https://user-images.githubusercontent.com/20972207/175783774-cb99f429-d1b8-4227-9c9d-05d1ed90d48f.png">

after:

<img width="821" alt="Screen Shot 2022-06-25 at 19 08 38" src="https://user-images.githubusercontent.com/20972207/175783771-5abb32cb-7b7f-455e-b55f-aa5232260fa3.png">

Reply:
This test is unrelated to the problem here. Could you revert these changes please?

Reply:
Yes!